activerecord/test/cases/autosave_association_test.rb

```diff
@@ -153,6 +154,24 @@ def test_should_not_add_the_same_callbacks_multiple_times_for_has_and_belongs_to
     assert_no_difference_when_adding_callbacks_twice_for Pirate, :parrots
   end
 
+  def test_should_save_associated_records_when_has_and_belongs_to_many_with_autosave
+    book = PublishedBook.new(name: "Good title", isbn: "010101101")
+    author = Author.create!(name: "Jhon")
+    book.authors_autosave << author
+    book.save!
+
+    like = Like.new(reason: "Good content.")
+    book = PublishedBook.find(book.id)
+    like.published_books << book
+    author = book.authors_autosave[0]
+    author.name = "Paul"
+    assert_equal true, author.changed?
+    assert_equal false, book.changed?
```

Comment:
Should give a little bit better message on failure

```suggestion
    assert_not_predicate book, :changed?
```